sync/class.jetpack-sync-queue.php

```diff
@@ -196,21 +196,36 @@ function checkout_with_memory_limit( $max_memory, $max_buffer_size = 500 ) {
 			OBJECT
 		);
 
+		if ( count( $items_with_size ) === 0 ) {
+			return false;
+		}
+
 		$total_memory = 0;
-		$item_ids     = array();
 
-		foreach ( $items_with_size as $item_with_size ) {
+		$min_item_id = $max_item_id = $items_with_size[0]->id;
+
+		foreach ( $items_with_size as $id => $item_with_size ) {
 			$total_memory += $item_with_size->value_size;
 
 			// if this is the first item and it exceeds memory, allow loop to continue
 			// we will exit on the next iteration instead
-			if ( $total_memory > $max_memory && count( $item_ids ) > 0 ) {
+			if ( $total_memory > $max_memory && $id > 0 ) {
 				break;
 			}
-			$item_ids[] = $item_with_size->id;
+
+			$max_item_id = $item_with_size->id;
 		}
 
-		$items = $this->fetch_items_by_id( $item_ids );
+		$query = $wpdb->prepare( 
+			"SELECT option_name AS id, option_value AS value FROM $wpdb->options WHERE option_name >= %s and option_name <= %s ORDER BY option_name ASC",
+			$min_item_id,
+			$max_item_id
```

Comment:
Do you expect any issues from using `>` and `<` comparisons with [how we currently name our options](https://github.com/Automattic/jetpack/blob/72caed3f9334e087ff435c315c28095aa32f367a/sync/class.jetpack-sync-queue.php#L343-L359)?